src/cmap/auth/gssapi.ts

```diff
@@ -30,60 +51,161 @@ export class GSSAPI extends AuthProvider {
       mechanismProperties['gssapiServiceName'] ||
       'mongodb';
 
-    const MongoAuthProcess = Kerberos.processes.MongoAuthProcess;
+    function externalCommand(
+      command: object,
+      cb: (
+        err: Error | MongoError | undefined,
+        result: { result: { payload: string; conversationId: any } }
+      ) => void
+    ) {
+      return connection.command('$external.$cmd', command, cb);
+    }
+
+    initialize(
+      {
+        retries: 10,
+        host,
```

Comment:
`host` and `port` should be properties on the `connection` that's passed in, is there a good reason to explicitly extract them and pass them around?